nutanix/resource_nutanix_ndb_profiles.go

```diff
@@ -0,0 +1,1280 @@
+				req.Properties = expandSoftwareProfileProp(ps.([]interface{}))
+			}
+
+			if ac, ok1 := d.GetOk("available_cluster_ids"); ok1 {
+				st := ac.([]interface{})
+				sublist := make([]*string, len(st))
+
+				for a := range st {
+					sublist[a] = utils.StringPtr(st[a].(string))
+				}
+				req.AvailableClusterIds = sublist
+			}
+		}
+	}
+
+	versionID := res.Versions[0].ID
+
+	_, eror := conn.Service.UpdateProfile(ctx, netReq, d.Id())
+	if eror != nil {
+		return diag.FromErr(eror)
+	}
+
+	_, er := conn.Service.UpdateProfileVersion(ctx, req, d.Id(), *versionID)
+	if er != nil {
+		return diag.FromErr(er)
+	}
+
+	return resourceNutanixNDBProfileRead(ctx, d, meta)
+}
+
+func resourceNutanixNDBProfileDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	resp, err := conn.Service.DeleteProfile(ctx, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if resp == utils.StringPtr("Profile Successfully Deleted.") {
+		d.SetId("")
+	}
+	return nil
+}
+
+func buildComputeProfileRequest(p interface{}) []*era.ProfileProperties {
+	if p != nil {
+		computeProp := []*era.ProfileProperties{}
+		pc := p.([]interface{})
+		for _, v := range pc {
+			val := v.(map[string]interface{})
+			if cpu, ok := val["cpus"]; ok {
+				computeProp = append(computeProp, &era.ProfileProperties{
+					Name:   utils.StringPtr("CPUS"),
+					Value:  utils.StringPtr(cpu.(string)),
+					Secure: false,
+				})
+			}
+
+			if coreCPU, ok := val["core_per_cpu"]; ok {
+				computeProp = append(computeProp, &era.ProfileProperties{
+					Name:   utils.StringPtr("CORE_PER_CPU"),
+					Value:  utils.StringPtr(coreCPU.(string)),
+					Secure: false,
+				})
+			}
+
+			if mem, ok := val["memory_size"]; ok {
+				computeProp = append(computeProp, &era.ProfileProperties{
+					Name:   utils.StringPtr("MEMORY_SIZE"),
+					Value:  utils.StringPtr(mem.(string)),
+					Secure: false,
+				})
+			}
+		}
+		return computeProp
+	}
+	return nil
+}
+
+func expandNetworkProfileProperties(ctx context.Context, meta interface{}, ps []interface{}) []*era.ProfileProperties {
+	prop := []*era.ProfileProperties{}
+	if len(ps) > 0 {
+		for _, v := range ps {
+			inst := v.(map[string]interface{})
+
+			if sIns, ok := inst["single_instance"]; ok && len(sIns.([]interface{})) > 0 {
+				prop = expandNetworkSingleInstance(sIns.([]interface{}))
+			}
+
+			if hIns, ok := inst["ha_instance"]; ok && len(hIns.([]interface{})) > 0 {
+				prop = expandNetworkHAInstance(ctx, meta, hIns.([]interface{}))
+			}
+		}
+	}
+	return prop
+}
+
+func buildDatabaseProfileProperties(ps []interface{}) []*era.ProfileProperties {
+	prop := []*era.ProfileProperties{}
+	if len(ps) > 0 {
+		for _, v := range ps {
+			val := v.(map[string]interface{})
+			if psdb, ok := val["postgres_database"]; ok {
+				brr := psdb.([]interface{})
+
+				for _, v := range brr {
+					val := v.(map[string]interface{})
+
+					if p1, ok1 := val["max_connections"]; ok1 {
```

Comment:
There are around more than 10 properties to be set. Instead of setting for each props, we can set it in a loop as most of them have same structure and types. 
```suggestion
					for key, val := range brr:
					        prop = append(prop, {name:key, value:val, secure: False})
					
```

Reply:
Something like this ^